demo/android/Activity/cobra-activity-demo-app/src/main/java/ai/picovoice/cobraactivitydemo/MainActivity.java

```diff
@@ -45,17 +47,28 @@
 
     private static final String APP_ID = "${YOUR_APP_ID_HERE}";
 
+    private ToggleButton recordButton;
+    private TextView errorMessage;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.cobra_activity_demo);
 
         voiceDetectedBackgroundColor = getResources().getColor(R.color.colorAccent);
 
+        recordButton = findViewById(R.id.startButton);
+        errorMessage = findViewById(R.id.errorMessage);
+
         try {
             cobra = new Cobra(APP_ID);
         } catch (CobraException e) {
-            displayError("Failed to initialize Cobra: \n" + e.toString());
+            String error = "Failed to initialize Cobra: \n" + e.getMessage();
```

Comment:
This still only shows a generic "Initialization failed" error, is there a way we can get more descriptive about what happened like you did with the ios demo?